app/org/sagebionetworks/bridge/services/ScheduledActivityService.java

```diff
@@ -157,16 +160,28 @@ public void updateScheduledActivities(String healthCode, List<ScheduledActivity>
             if (schActivity.getGuid() == null) {
                 throw new BadRequestException(String.format("Task #%s has no GUID", i));
             }
+            if (byteLength(schActivity.getClientData()) > CLIENT_DATA_MAX_BYTES) {
+                throw new BadRequestException("Client data too large ("+CLIENT_DATA_MAX_BYTES+" bytes limit)");
+            }
+            ScheduledActivity dbActivity = activityDao.getActivity(healthCode, schActivity.getGuid());
+            boolean addToSaves = false;
+            if (hasUpdatedClientData(schActivity, dbActivity)) {
+                dbActivity.setClientData(schActivity.getClientData());
+                addToSaves = true;
+            }
             if (schActivity.getStartedOn() != null || schActivity.getFinishedOn() != null) {
                 // We do not need to add the time zone here. Not returning these to the user.
```

Comment:
Yes, now that we are tracking when to save, outer if statement isn't needed. Removed.